txscript/script.go

```diff
@@ -410,7 +410,7 @@ func calcHashOutputs(tx *wire.MsgTx) wire.ShaHash {
 // being spent, in addition to the final transaction fee. In the case the wallet
 // if fed an invalid input amount, the real sighash will differ causing the
 // produced signature to be invalid.
-func calcWitnessSignatureHash(subScript []parsedOpcode, sigHashes *TxSigHashes,
+func CalcWitnessSignatureHash(subScript []parsedOpcode, sigHashes *TxSigHashes,
```

Comment:
As we discussed on chat, rather than just publicly exporting this function, a publicly exported wrapper function (which simply calls the private function) should be created instead. With this route, we can avoid publicly exporting an object which has no public attributes or methods.